core/src/main/java/com/datastrato/gravitino/listener/impl/AsyncQueueListener.java

```diff
@@ -0,0 +1,133 @@
+  private final AtomicLong dropEventCounters = new AtomicLong(0);
+  private final AtomicLong lastDropEventCounters = new AtomicLong(0);
+  private Instant lastRecordDropEventTime;
+  private final String asyncQueueListenerName;
+
+  public AsyncQueueListener(List<EventListenerPlugin> listeners, String name, int queueCapacity) {
+    this.asyncQueueListenerName = NAME_PREFIX + name;
+    this.eventListeners = listeners;
+    this.queue = new LinkedBlockingQueue<>(queueCapacity);
+    this.asyncProcessor = new Thread(() -> processEvents());
+    asyncProcessor.setDaemon(true);
+    asyncProcessor.setName(asyncQueueListenerName);
+  }
+
+  @Override
+  public void onPostEvent(Event event) {
+    if (stopped.get()) {
+      LOG.warn(
+          "{} drop event: {}, since AsyncQueueListener is stopped",
+          asyncQueueListenerName,
+          event.getClass().getSimpleName());
+      return;
+    }
+
+    if (queue.offer(event)) {
+      return;
+    }
+
+    long currentDropEvents = dropEventCounters.incrementAndGet();
+    long lastDropEvents = lastDropEventCounters.get();
+    // dropEvents may less than zero in such conditions:
+    // 1. Thread A increment dropEventCounters
+    // 2. Thread B increment dropEventCounters and update lastDropEventCounters
+    // 3. Thread A get lastDropEventCounters
+    long dropEvents = currentDropEvents - lastDropEvents;
+    if (dropEvents > 0 && Instant.now().isAfter(lastRecordDropEventTime.plusSeconds(60))) {
+      if (lastDropEventCounters.compareAndSet(lastDropEvents, currentDropEvents)) {
+        LOG.warn(
+            "{} drop {} events since {}",
+            asyncQueueListenerName,
+            dropEvents,
+            lastRecordDropEventTime);
+        lastRecordDropEventTime = Instant.now();
+      }
+    }
+  }
+
+  @Override
+  public void init(Map<String, String> properties) {
+    throw new RuntimeException(
+        "Should not reach here, the event listener has already been initialized.");
+  }
+
+  @Override
+  public void start() {
+    eventListeners.forEach(listenerPlugin -> listenerPlugin.start());
+    asyncProcessor.start();
+  }
+
+  @Override
+  public void stop() {
+    Preconditions.checkState(!stopped.get(), asyncQueueListenerName + " had already stopped");
+    stopped.compareAndSet(false, true);
+    asyncProcessor.interrupt();
+    try {
+      asyncProcessor.join(3000);
+    } catch (InterruptedException e) {
+      LOG.warn("{} interrupt async processor failed.", asyncQueueListenerName, e);
+    }
+    eventListeners.forEach(listenerPlugin -> listenerPlugin.stop());
+  }
+
+  @VisibleForTesting
+  List<EventListenerPlugin> getEventListeners() {
+    return this.eventListeners;
+  }
+
+  private void processEvents() {
+    while (Thread.currentThread().isInterrupted() == false) {
+      try {
+        Event event = queue.take();
+        this.eventListeners.stream().forEach(listener -> listener.onPostEvent(event));
```

Comment:
The exception of one listener will affect the following listener, I think you should try...catch per listener.

Reply:
exception is handled in `EventListenerPluginWrapper`